lib/matplotlib/backend_bases.py

```diff
@@ -2130,6 +2130,8 @@ def print_figure(self, filename, dpi=None, facecolor=None, edgecolor=None,
 
         if format is None:
             # get format from filename, or from backend's default filetype
+            if isinstance(filename, getattr(os, "PathLike", ())):
```

Comment:
Am I correct, that this will only fix the format detection for Python 3.6+? Since this will probably go into 2.2.x. it should also hande pre-python3.6 Path objects, i.e. check for `Path` instance and use `filename = str(filename)`?

Reply:
Also [PEP-519 ](https://www.python.org/dev/peps/pep-0519/#backwards-compatibility) states

> Libraries wishing to support path objects and a version of Python prior to Python 3.6 and the existence of `os.fspath()` can use the idiom of `path.__fspath__() if hasattr(path, "__fspath__") else path`.

But that's more a matter of taste.

Reply:
The original PR explicitly only supported 3.6+ (see whatsnew note in https://github.com/matplotlib/matplotlib/pull/10231/files).  I have no intent to reimplement the pathlike API on <3.6.

So I'll leave things as they are.